docs/spark/how-to-guides/batch-processing.md

```diff
@@ -0,0 +1,129 @@
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName("GitHub and Spark Batch")
+        .GetOrCreate();
+   ```
+
+1. Read the input file into a DataFrame, which is a distributed collection of data organized into named columns. You can set the columns for your data through `.Schema()`. Use `.Show()` to display the data in your DataFrame.
+
+   ```csharp
+   DataFrame projectsDf = spark
+       .Read()
+       .Schema("id INT, url STRING, owner_id INT, " +
+       "name STRING, descriptor STRING, language STRING, " +
+       "created_at STRING, forked_from INT, deleted STRING" +
+       "updated_at STRING")
+       .Csv(args[0]);
+
+   projectsDf.Show();
+   ```
+
+1. Use `DataFrame.Na()` to drop rows with NA (null) values, and the `DataFrameNaFunctions.Drop()` method to remove certain columns from your data. This helps prevent errors if you try to analyze null data or columns that are not relevant to your final analysis.
+
+   ```csharp
+   // Drop any rows with NA values
+   DataFrameNaFunctions dropEmptyProjects = projectsDf.Na();
+   DataFrame cleanedProjects = dropEmptyProjects.Drop("any");
+
+   // Remove unnecessary columns
+   cleanedProjects = cleanedProjects.Drop("id", "url", "owner_id");
+   cleanedProjects.Show();
+   ```
+
+1. Spark SQL allows you to make SQL calls on your data. It's common to combine user-defined functions and Spark SQL to apply a user-defined function to all rows of your DataFrame.
+
+   You can specifically call `spark.Sql` to mimic standard SQL calls seen in other types of apps. You can also call methods like `GroupBy` and `Agg` to specifically combine, filter, and perform calculations on your data.
+
+   ```csharp
+   // Average number of times each language has been forked
+   DataFrame groupedDF = cleanedProjects
+       .GroupBy("language")
+       .Agg(Avg(cleanedProjects["forked_from"]) 
+
+   // Sort by most forked languages first
+   groupedDF.OrderBy(Desc("avg(forked_from)")).Show(); 
+
+   spark.Udf().Register<string, bool>(
+       "MyUDF",
+       (date) => DateTime.TryParse(date, out DateTime   convertedDate) &&
+           (convertedDate > s_referenceDate);   
+   cleanedProjects.CreateOrReplaceTempView("dateView"); 
+
+   DataFrame dateDf = spark.Sql(
+       "SELECT *, MyUDF(dateView.updated_at) AS datebefore FROM dateView");
+   dateDf.Show();
+   ```
+
+1. Call `spark.Stop()` to end the SparkSession.
+
+1. Use `spark-submit` to run your app. Be sure to update the following command with the actual paths to your Microsoft Spark jar file and **projects_smaller.csv**.
+
+
+   ```console
+   spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /<path>/to/microsoft-spark-<version>.jar Microsoft.Spark.CSharp.Examples.exe Sql.Batch.GitHubProjects /<path>/projects_smaller.csv
```

Comment:
I realize this command is specifically running apps when they come from the GitHub repo (when they're in the Microsoft.Spark.CSharp.Examples folder and exe). Let's change to something more general:

```suggestion
   spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /<path>/to/microsoft-spark-<version>.jar dotnet /<path>/to/netcoreapp<version>/GitHubProjects.dll /<path>/projects_smaller.csv
```

Reply:
We should also add a bit more info to this submission/running step, something like the following:

-Build your app: `dotnet build`
-Run your app via `spark-submit`: spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner --master local /<path>/to/microsoft-spark-<version>.jar dotnet /<path>/to/netcoreapp<version>/GitHubProjects.dll /<path>/projects_smaller.csv

For more instructions about running your .NET for Spark app, check out the guides for [building on Windows](https://github.com/dotnet/spark/blob/master/docs/building/windows-instructions.md#using-net-core-cli-for-net-core-21x) or [Building on Ubuntu](https://github.com/dotnet/spark/blob/master/docs/building/ubuntu-instructions.md).